### PR TITLE
XW-2897 | ooyala player html5-skin DS icons

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-2897

@Wikia/x-wing

### CODE REVIEW

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/ArticleVideo/OoyalaConfigController.class.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+				'fontFamilyName' => 'ooyala-slick-type',
+				'fontString' => 'd',
+				'fontStyleClass' => 'oo-icon oo-icon-system-auto',
+			],
+		],
+	];
+
+	public function skin() {
+		$config = self::CONFIG;
+		$config['icons']['play']['svg'] = DesignSystemHelper::renderSvg('wds-icons-play-triangle-small');
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/OoyalaConfigController.class.php]: can we render those svgs when defining constant?

File: extensions/wikia/ArticleVideo/OoyalaConfigController.class.php [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+		$config['icons']['pause']['svg'] = DesignSystemHelper::renderSvg('wds-icons-pause-small');
+		$config['icons']['volume']['svg'] = DesignSystemHelper::renderSvg('wds-icons-volume-small');
+		$config['icons']['volumeOff']['svg'] = DesignSystemHelper::renderSvg('wds-icons-volume-off-small');
+		$config['icons']['expand']['svg'] = DesignSystemHelper::renderSvg('wds-icons-fullscreen-small');
+		$config['icons']['compress']['svg'] = DesignSystemHelper::renderSvg('wds-icons-fullscreen-off-small');
+		$config['icons']['share']['svg'] = DesignSystemHelper::renderSvg('wds-icons-share-small');
+		$this->getResponse()->setData( $config );
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/OoyalaConfigController.class.php]: I think we should define some caching rules for this controller

Comment (kvas-damian) [reply extensions/wikia/ArticleVideo/OoyalaConfigController.class.php]: `$this->response->setCacheValidity()`

File: extensions/wikia/ArticleVideo/OoyalaConfigController.class.php [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+			],
+			'auto' => [
+				'fontFamilyName' => 'ooyala-slick-type',
+				'fontString' => 'd',
+				'fontStyleClass' => 'oo-icon oo-icon-system-auto',
+			],
+		],
+	];
+
+	public function skin() {
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/OoyalaConfigController.class.php]: This method doesn't have template that could be rendered. Can we enforce `json` format then?

File: extensions/wikia/ArticleVideo/styles/_video-controls.scss [hunk 4]
```diff
@@ -77,67 +77,37 @@
...
 			}
-		}
-
-		.fandom-video-gear-icon {
-			background: no-repeat center url($gear-icon-uri);
 
 			&:hover {
-				background-image: url($gear-icon-uri-hover);
+				svg {
+					fill: $video-player-blue !important;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/styles/_video-controls.scss]: Why do we need `!important`?
Do we want to apply the same color to all svgs?

Comment (latata) [reply extensions/wikia/ArticleVideo/styles/_video-controls.scss]: We don't need it. ;) will fix that
And we want to fill all ooyala icons svgs on hover

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 define('ooyala-player', function () {
 
-	var baseJSONSkinUrl = '/extensions/wikia/ArticleVideo/scripts/ooyala/skin.json?v=2';
+	var baseJSONSkinUrl = '/wikia.php?controller=OoyalaConfig&method=skin&' + window.wgStyleVersion;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js]: we try to follow pattern `'cb=' + window.wgStyleVersion`

File: extensions/wikia/ArticleVideo/styles/_video-controls.scss [hunk 6]
```diff
@@ -77,67 +77,37 @@
...
-				background-image: url($volume-icon-uri-hover);
-			}
+		.oo-icon:hover {
+			color: $video-player-blue !important;
 		}
 
-		.fandom-video-volume-off-icon {
-			background: no-repeat center url($volume-off-icon-uri);
+		.oo-icon-svg {
+			filter: none !important;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/styles/_video-controls.scss]: what about other `important`s. Are they required?

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 define('ooyala-player', function () {
 
-	var baseJSONSkinUrl = '/extensions/wikia/ArticleVideo/scripts/ooyala/skin.json?v=2';
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js]: should we remove this file now?

File: extensions/wikia/ArticleVideo/styles/_video-controls.scss (hunk 6 above)
Comment (latata) [reply extensions/wikia/ArticleVideo/styles/_video-controls.scss]: They are because html5-skin adds inline styles there

**Review by rogatty — APPROVED** [review]

**Pull request merged**